syntax = "proto3";

package admin_lesson;

service AdminLessonService {
    rpc AdminCreateLesson (AdminCreateLessonRequest) returns (AdminCreateLessonResponse);
    rpc AdminGetLesson    (AdminGetLessonRequest)    returns (AdminGetLessonResponse);
    rpc AdminUpdateLesson (AdminUpdateLessonRequest) returns (AdminUpdateLessonResponse);
    rpc AdminDeleteLesson (AdminDeleteLessonRequest) returns (AdminDeleteLessonResponse);
    rpc AdminListLessons  (AdminListLessonsRequest)  returns (AdminListLessonsResponse);
}

message AdminCreateLessonRequest {
    string module_id = 1;
    string title     = 2;

    string type = 3;
    string access_type = 4;

    optional string content   = 5;
    optional string video_id  = 6;
    optional string duration  = 7;

    optional int32 order_number = 8;
    bool is_active = 9;
}

message AdminCreateLessonResponse {
    string id = 1;
}

message AdminGetLessonRequest {
    string id = 1;
}

message AdminGetLessonResponse {
    string id        = 1;
    string module_id = 2;
    string title     = 3;

    string type = 4;
    string access_type = 5;

    optional string content   = 6;
    optional string video_id  = 7;
    optional string duration  = 8;

    int32 order_number = 9;
    bool is_active = 10;
}

message AdminUpdateLessonRequest {
    string id = 1;

    optional string module_id = 2;
    optional string title     = 3;

    string type = 4;
    string access_type = 5;

    optional string content   = 6;
    optional string video_id  = 7;
    optional string duration  = 8;

    optional int32 order_number = 9;
    optional bool is_active = 10;
}

message AdminUpdateLessonResponse {
    string id = 1;
}

message AdminDeleteLessonRequest {
    string id = 1;
}

message AdminDeleteLessonResponse {
    bool success = 1;
}

message AdminListLessonsRequest {
    optional string module_id = 1;
}

message AdminListLessonsResponse {
    repeated AdminGetLessonResponse items = 1;
}
